libsql-server/src/namespace/configurator/helpers.rs

```diff
@@ -328,8 +329,31 @@ where
             line = tokio::task::spawn_blocking({
                 let conn = conn.clone();
                 move || -> crate::Result<String, LoadDumpError> {
-                    conn.with_raw(|conn| conn.execute(&line, ())).map_err(|e| {
-                        LoadDumpError::Internal(format!("line: {}, error: {}", line_id, e))
+                    conn.with_raw(|conn| {
+                        conn.authorizer(Some(|auth: AuthContext<'_>| match auth.action {
+                            AuthAction::Attach { filename: _ } => Authorization::Deny,
+                            _ => Authorization::Allow,
+                        }));
+                        conn.execute(&line, ())
+                    })
+                    .map_err(|e| match e {
+                        rusqlite::Error::SqlInputError {
+                            msg, sql, offset, ..
+                        } => {
+                            let msg = if sql.to_lowercase().contains("attach") {
```

Comment:
Can we instead check error code from `SqlInputError` and compare it with something like `AuthorizationForStatementDenied`?

Reply:
I was thinking that but when I call https://docs.rs/rusqlite/latest/rusqlite/enum.Error.html#method.sqlite_error  and the other one on `Error` it returns both `None` so I am not totally sure what is going on.

Reply:
Ah, sad situation :(

Ok, let's leave it as is then!